mkosi/distributions/gentoo.py

```diff
@@ -247,8 +248,8 @@ def install(cls, state: MkosiState) -> None:
         if state.config.make_initrd:
             return
 
-        invoke_emerge(state, sysroot=cls.stage3_cache, opts=opts,
-                      pkgs=["sys-kernel/gentoo-kernel-bin"])
+        if state.config.bootable == ConfigFeature.enabled and not state.staging.joinpath(state.config.output_split_uki).exists():
+            invoke_emerge(state, sysroot=cls.stage3_cache, opts=opts, pkgs=["sys-kernel/gentoo-kernel-bin"])
```

Comment:
We do not install a kernel even if bootable is specified, again, please look at the other distributions, none of them do this, gentoo shouldn't do it either.

Reply:
that would violate POLA, but sure.

Reply:
so on the POLA violation:

we're really asking the user to go down the trial-and-error to find out that if he wants a bootable image he needs to do 2 things:
1. add `gnuefi` to `pkgmngr`
2. add `systemd` to `Packages=`. (actually `sys-apps/systemd` see below)

do we really want this? i for one can't think of a clean workaround for this. for instance, we check in `invoke_emerge()` to to see if users wants a bootable image and `systemd` was asked to be merged in `packages`. this is messy and error prone, for one `systemd` could mean `dev-ml/systemd`.

Reply:
so we enforce `gnuefi` in `invoke_emerge()` and that should fix it. now let me check the CI failures.